test/v1/controllers/RetrieveEOPSObligationsControllerSpec.scala

```diff
@@ -0,0 +1,231 @@
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.handleRequest(
+              nino = nino,
+              typeOfBusiness = Some(typeOfBusiness.toString),
+              incomeSourceId = Some(incomeSourceId),
+              fromDate = Some(fromDate),
+              toDate = Some(toDate),
+              status = Some(status.toString)
+            )(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (TypeOfBusinessFormatError, BAD_REQUEST),
+          (BusinessIdFormatError, BAD_REQUEST),
+          (FromDateFormatError, BAD_REQUEST),
+          (ToDateFormatError, BAD_REQUEST),
+          (StatusFormatError, BAD_REQUEST),
+          (MissingFromDateError, BAD_REQUEST),
+          (MissingToDateError, BAD_REQUEST),
+          (ToDateBeforeFromDateError, BAD_REQUEST),
+          (MissingTypeOfBusinessError, BAD_REQUEST),
+          (RuleDateRangeInvalidError, BAD_REQUEST),
+          (RuleFromDateNotSupportedError, BAD_REQUEST),
+          (BadRequestError, BAD_REQUEST)
+        )
+
+        input.foreach(args => (errorsFromParserTester _).tupled(args))
+      }
+
+      "service errors occur" must {
+        def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
+          s"a ${mtdError.code} error is returned from the service" in new Test {
+
+            MockRetrieveEOPSObligationsRequestParser
+              .parse(rawData)
+              .returns(Right(requestData))
+
+            MockRetrieveEOPSObligationsService
+              .retrieve(requestData)
+              .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), mtdError))))
+
+            val result: Future[Result] = controller.handleRequest(
+              nino = nino,
+              typeOfBusiness = Some(typeOfBusiness.toString),
+              incomeSourceId = Some(incomeSourceId),
+              fromDate = Some(fromDate),
+              toDate = Some(toDate),
+              status = Some(status.toString)
+            )(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(mtdError)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (BusinessIdFormatError, BAD_REQUEST),
+          (FromDateFormatError, BAD_REQUEST),
+          (ToDateFormatError, BAD_REQUEST),
+          (StatusFormatError, BAD_REQUEST),
+          (RuleDateRangeInvalidError, BAD_REQUEST),
+          (NotFoundError, NOT_FOUND),
+          (NoObligationsFoundError, NOT_FOUND),
+          (DownstreamError, INTERNAL_SERVER_ERROR),
+        )
```

Comment:
These errors are not the same as the errors in the spec

Reply:
Ah yeah I was following the green ticks, not the tech spec - changed!